File: output/StarCoder/avatar/Python/Java/atcoder_ABC147_D.java
import java.util.*;
import java.io.*;
public class atcoder_ABC147_D {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        long[] A = new long[N];
        for (int i = 0; i < N; i++) {
            A[i] = sc.nextLong();
        }
        long result = 0;
        for (int b = 0; b < 60; b++) {
            long bs = 0;
            for (int i = 0; i < N; i++) {
                bs += (A[i] >> b) & 1;
            }
            result = (result + bs * (N - bs) * (1 << b)) % 1000000007;
        }
        System.out.println(result);
    }
}

File: output/StarCoder/avatar/Java/Python/atcoder_ABC147_D.py
import java.util.*;
public class atcoder_ABC147_D {
    public static void main(String[] args) {
        long startTime = System.currentTimeMillis();
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        long div = (long) Math.pow(10, 9) + 7;
        long[] A = new long[n];
        long ans = 0;
        for (int i = 0; i < n; i++) {
            A[i] = sc.nextLong();
        }
        for (int i = 0; i < 60; i++) {
            long a = 0;
            long count1 = 0;
            for (int j = 0; j < n; j++) {
                if ((A[j] >> i & 1) == 1) count1++;
            }
            a = a + count1 * (n - count1);
            for (int j = 0; j < i; j++) {
                a = (a << 1) % div;
            }
            ans += a;
            ans %= div;
        }
        System.out.println(ans);
        long endTime = System.currentTimeMillis();
    }
}